### PR TITLE
[GitHub] Fix scopes related issue

### PR DESCRIPTION
**Pull request by parthpuri-elastic**
## Closes https://github.com/elastic/connectors/issues/1983

Connector throws an error in the case of admin:org and write:org scopes. Now add admin:org and write:org to valid scopes and add a warning if the connector finds higher privileges than required.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

**Comment by moxarth-elastic** [comment]
buildkite test this

**Comment by moxarth-elastic** [comment]
buildkite test this

**Review by wangch079 — COMMENTED** [review]
It seems `write:org` and `admin:org` will auto select `read:org`:
image

image

**Comment by parthpuri-elastic** [comment]
> It seems `write:org` and `admin:org` will auto select `read:org`: image
> 
> image

Yes, but in the response, we are not getting `read:org` and connector checking `read:org` only. So, i have append `read:org` when we are getting `admin:org` and `write:org`.

File: tests/sources/test_github.py [hunk 1]
```diff
@@ -600,6 +600,18 @@ async def test_validate_config_with_invalid_token_then_raise():
...
+    async with create_github_source() as source:
+        source.github_client.post = AsyncMock(
+            return_value=(
+                {"user": "username"},
+                {"X-OAuth-Scopes": "user, repo, admin:org"},
+            )
+        )
+        assert await source.validate_config() is None
+
+
```
Comment (wangch079) [inline tests/sources/test_github.py:614]: Consider adding more test cases when `ConfigurableFieldValueError` is raised.

**Comment by moxarth-elastic** [comment]
buildkite test this

File: tests/sources/test_github.py [hunk 2]
```diff
@@ -587,11 +587,15 @@ async def test_ping_with_unsuccessful_connection():
 
 
 @pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "scopes",
+    ["", "repo", "manage_runner:org, delete:packages, admin:public_key"],
+)
 @patch("connectors.utils.time_to_sleep_between_retries", Mock(return_value=0))
-async def test_validate_config_with_invalid_token_then_raise():
+async def test_validate_config_with_invalid_token_then_raise(scopes):
```
Comment (wangch079) [inline tests/sources/test_github.py:595]: The test case does not tally with the function name. This is supposed to test with a invalid token (the `pytest.raises` also imply the exception message is about invalid token)

Could you have a separate test case like `test_validate_config_with_insufficient_scope`?

File: tests/sources/test_github.py (hunk 1 above)
Comment (parthpuri-elastic) [reply tests/sources/test_github.py:618]: added testcases

**Comment by moxarth-elastic** [comment]
buildkite test this

File: tests/sources/test_github.py (hunk 2 above)
Comment (parthpuri-elastic) [reply tests/sources/test_github.py]: Sure, added separate test case `test_validate_config_with_insufficient_scope` and validated `UnauthorizedException` for invalid token

File: tests/sources/test_github.py [hunk 3]
```diff
@@ -600,6 +615,18 @@ async def test_validate_config_with_invalid_token_then_raise():
...
+@pytest.mark.asyncio
+async def test_validate_config_with_extra_scopes_token():
+    async with create_github_source() as source:
+        source.github_client.post = AsyncMock(
+            return_value=(
+                {"user": "username"},
+                {"X-OAuth-Scopes": "user, repo, admin:org"},
+            )
+        )
+        assert await source.validate_config() is None
```
Comment (wangch079) [inline tests/sources/test_github.py:627]:
```suggestion
async def test_validate_config_with_extra_scopes_token(patch_logger):
    async with create_github_source() as source:
        source.github_client.post = AsyncMock(
            return_value=(
                {"user": "username"},
                {"X-OAuth-Scopes": "user, repo, admin:org"},
            )
        )
        assert await source.validate_config()
        patch_logger.assert_present("Configured token does not have required rights to fetch the content. Required scopes are 'repo', 'user', and 'read:org'.")
```

File: connectors/sources/github.py [hunk 4]
```diff
@@ -1196,9 +1196,25 @@ async def _remote_validation(self):
         _, headers = await self.github_client.post(  # pyright: ignore
             query_data=query_data, need_headers=True
         )
-        if "repo" not in headers.get("X-OAuth-Scopes"):  # pyright: ignore
-            msg = "Configured token does not have required rights to fetch the content"
+
+        scopes = headers.get("X-OAuth-Scopes", "")
+        scopes = [scope.strip() for scope in scopes.split(",")]
+        required_scopes = ["repo", "user", "read:org"]
```
Comment (wangch079) [inline connectors/sources/github.py:1202]: Nit, why not define `required_scopes` as a set, as I see it's always used as `set(required_scopes)`. You can also consider to define `scopes` as a set as well.

File: tests/sources/test_github.py (hunk 3 above)
Comment (parthpuri-elastic) [reply tests/sources/test_github.py]: Addressed in https://github.com/elastic/connectors/pull/1997/commits/6f3db163890fc46f2355bdcf0be878c67a679f83

**Review by wangch079 — APPROVED** [review]
Nice work

**Pull request merged**